### PR TITLE
sci-physics/thepeg: new package, add 2.3.0

### PR DESCRIPTION
**Pull request by APN-Pucky**
PR created by [pkgpr](https://github.com/APN-Pucky/pkgpr).

### CODE REVIEW

**Comment by APN-Pucky** [comment]
This package does already exist in `::gentoo` but the new version depends on `sci-physics/rivet` that is only available in `::sci`. I'd prefer to move rivet to `::gentoo`, but I thought I'd ask here first what is better?

**Comment by AndrewAmmerlaan** [comment]
> This package does already exist in `::gentoo` but the new version depends on `sci-physics/rivet` that is only available in `::sci`. I'd prefer to move `rivet` to `::gentoo`, but I thought I'd ask here first what is better?

The `rivet` ebuild looks fine (haven't tested it though), so I think we can move this to `::gentoo`. We should check if we can add `python3_12` though, and it looks like it should inherit `fortran-2.eclass` (we can then drop the explicit dependency on `virtual/fortran`.

File: sci-physics/thepeg/thepeg-2.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+CDEPEND="
+	sci-libs/gsl:0=
+	emacs? ( >=app-editors/emacs-23.1:* )
+	fastjet? ( sci-physics/fastjet:0= )
+	rivet? ( sci-physics/rivet:3=[hepmc3] )
+	hepmc3? ( sci-physics/hepmc:3= )
+	lhapdf? ( >=sci-physics/lhapdf-6.0:0= )
+	zlib? ( sys-libs/zlib:0= )"
+DEPEND="${CDEPEND}
+	sci-libs/gsl
```
Comment (AndrewAmmerlaan) [inline sci-physics/thepeg/thepeg-2.3.0.ebuild:38]:
```suggestion
	sci-libs/gsl:=
```

File: sci-physics/thepeg/thepeg-2.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+src_test() {
+	emake LHAPATH="${DISTDIR}" check
+}
+
+src_install() {
+	default
+	use emacs && elisp-install ${PN} lib/ThePEG.el{,c}
+	use java && java-pkg_newjar java/ThePEG.jar
+
+	cat <<-EOF > "${T}"/50${PN}
```
Comment (AndrewAmmerlaan) [inline sci-physics/thepeg/thepeg-2.3.0.ebuild:94]:
```suggestion
	cat <<-EOF > "${T}"/50${PN} || die
```

File: sci-physics/thepeg/thepeg-2.3.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	zlib? ( sys-libs/zlib:0= )"
+DEPEND="${CDEPEND}
+	sci-libs/gsl
+	java? ( virtual/jdk:1.8 )
+	test? (
+		sys-process/time
+		dev-libs/boost
+	)"
+RDEPEND="${CDEPEND}
+	java? ( virtual/jre:1.8 )"
```
Comment (AndrewAmmerlaan) [inline sci-physics/thepeg/thepeg-2.3.0.ebuild:45]:
```suggestion
	java? ( virtual/jre:1.8 )
"
```

Can it be used with newer java? 8 is quite old.

Comment (APN-Pucky) [reply sci-physics/thepeg/thepeg-2.3.0.ebuild:45]: I haven't used it with it with java, this is still from the ebuilds in `::gentoo`.

**Comment by APN-Pucky** [comment]
> > This package does already exist in `::gentoo` but the new version depends on `sci-physics/rivet` that is only available in `::sci`. I'd prefer to move `rivet` to `::gentoo`, but I thought I'd ask here first what is better?
> 
> The `rivet` ebuild looks fine (haven't tested it though), so I think we can move this to `::gentoo`. We should check if we can add `python3_12` though, and it looks like it should inherit `fortran-2.eclass` (we can then drop the explicit dependency on `virtual/fortran`.

`rivet` even works well in prefixed gentoo. I added a docker image with startprefix entrypoint to also test prefix compatibility recently.

**Comment by APN-Pucky** [comment]
Moved to https://github.com/gentoo/gentoo/pull/36341

**Pull request closed without merging**